app/views/assignments/_form.html.slim

```diff
@@ -1,34 +1,26 @@
 = simple_form_for(@assignment) do |f|
   = simple_error_notice f
 
-  - if policy(Assignment).supervisor?
-    .row
-      .col-xs-12
-        - if action_name == 'new' && !params[:client_id]
-          = f.association :client, include_blank: true, collection: Client.need_accompanying
-        - else
-          = f.association :client, include_blank: true,
-            selected: params[:client_id] || (@assignment.client.id if @assignment.client.present?)
-
-    .row
-      .col-xs-12
-        = f.input :period, as: :date_range_picker, label: false
-    .row
-      .col-xs-12
-        - if action_name == 'new' && !params[:volunteer_id]
-          = f.association :volunteer, include_blank: true, collection: Volunteer.seeking_clients
-        - else
-          = f.association :volunteer, include_blank: true,
-            selected: params[:volunteer_id] || (@assignment.volunteer.id if @assignment.volunteer.present?)
-
-    .row
-      .col-xs-12
-        = f.input :state, collection: Assignment::STATES, default: :suggested
-
   .row
     .col-xs-12
-      = f.input :confirmation
+      - if action_name == 'new' && !params[:client_id]
+        = f.association :client, include_blank: true, collection: Client.need_accompanying
+      - else
+        = f.association :client, include_blank: true,
+          selected: params[:client_id] || (@assignment.client.id if @assignment.client.present?)
```

Comment:
`params[:client_id] || @assignment&.client.id`

Reply:
Combined with the next suggested change that results in validation error, `Client must exist, Volunteer must exist.`

Reply:
what is with this?
```ruby
f.association ... selected: @assignment&.client.id || params[:client_id]
```

would that work?

Reply:
Nope. `undefined method `id' for nil:NilClass`

Reply:
`params[:volunteer_id] || @assignment&.volunteer&.id`

that works. I tryed it. `@assignment&.volunteer.id` throws `NoMethodError: undefined method `id' for nil:NilClass` but `@assignment&.volunteer&.id` doesn't